client/components/community-translator/translatable.jsx

```diff
@@ -0,0 +1,214 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { isEmpty } from 'lodash';
+import { translate } from 'i18n-calypso';
+import classNames from 'classnames';
+import Gridicon from 'gridicons';
+
+/**
+ * Internal dependencies
+ */
+import Dialog from 'components/dialog';
+import Button from 'components/button';
+import TranslatableField from './translatable-field';
+import TranslatedSuccess from './translated-success';
+import { getTranslationData, getTranslationPermaLink, submitTranslation } from './utils.js';
+
+export class Translatable extends Component {
+	state = {
+		showDialog: false,
+		originalData: {},
+		formState: {},
+	};
+
+	hasDataLoaded() {
+		return ! isEmpty( this.state.originalData );
+	}
+
+	handleTranslationChange = event => {
+		const { name, value } = event.target;
+		this.setState( {
+			formState: {
+				...this.state.formState,
+				[ name ]: value,
+			},
+			submitting: false,
+			submissionSuccess: false,
+		} );
+	};
+
+	closeDialog = () => {
+		this.setState( {
+			showDialog: false,
+			submissionSuccess: false,
+			formState: this.state.submissionSuccess ? {} : this.state.formState,
+			originalData: this.state.submissionSuccess ? {} : this.state.originalData,
+		} );
+	};
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( { showDialog: true } );
+
+		const { singular, context, plural, locale } = this.props;
+		! this.hasDataLoaded() &&
+			getTranslationData( locale, { singular, context, plural } ).then( originalData =>
+				this.setState( {
+					error: originalData.error,
+					originalData,
+					translationUrl: getTranslationPermaLink( originalData.originalId, locale ),
```

Comment:
Nitpick: 
It's highly unlikely, but we should take `null` into consideration since that's one possible return value from `getTranslationPermaLink()`.  It currently results in an empty anchor like this:
![image](https://user-images.githubusercontent.com/1842898/38608700-a751adc4-3dae-11e8-8b4e-c64058a933f6.png)

Actually we could consider to keep it this way for now, because it could be quite tricky itself either to find a good default or to consider this case an error or not.

Reply:
Good find. Or maybe not render the link at all?

Reply:
Ideally I think we should show something more precise, since in this case the translation actually hasn't been submitted successfully.

To not render the link could be the first good step though, since this should be rare and would mostly be seen by us.